File: Project_Three/spring-project/web-service-operations/src/main/java/com/ozalp/entity_management_spring/business/impls/WebOperationServiceImpl.java
package com.ozalp.entity_management_spring.business.impls;

import com.ozalp.entity_management_spring.business.mappers.WebServiceOperationMapper;
import com.ozalp.entity_management_spring.business.service.WebOperationService;
import com.ozalp.entity_management_spring.dataAccess.WebServiceOperationRepository;
import com.ozalp.entity_management_spring.models.dtos.requests.CreateWebServiceOperationRequest;
import com.ozalp.entity_management_spring.models.dtos.responses.WebServiceOperationResponse;
import com.ozalp.entity_management_spring.models.entities.WebServiceOperation;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class WebOperationServiceImpl implements WebOperationService {

    private final WebServiceOperationRepository repository;
    private final WebServiceOperationMapper mapper;

    @Override
    public void add(CreateWebServiceOperationRequest request) {
        WebServiceOperation reqEntity = mapper.toEntity(request);
        repository.save(reqEntity);
    }

    @Override
    public void delete(String name) {
        List<WebServiceOperation> myEntityList = repository.findByName(name);

        for (WebServiceOperation entity : myEntityList) {
            repository.delete(entity);
        }
    }

    @Override
    public List<WebServiceOperationResponse> getAll() {
        return repository.findAll()
                .stream()
                .map(mapper::toResponse)
                .toList();
    }

    @Override
    public List<WebServiceOperationResponse> search(String name) {
        return repository.findByName(name).stream()
                .map(mapper::toResponse)
                .toList();
    }
}

File: Project_Two/mvc/EntityManagementAspNet/EntityManagementAspNet.Business/Impl/WebOperationServiceImpl.cs
using EntityManagementAspNet.Business.Dtos;
using EntityManagementAspNet.Business.Service;
using EntityManagementAspNet.DataAccess.Abstract;
using EntityManagementAspNet.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EntityManagementAspNet.Business.Impl
{
    public class WebOperationServiceImpl: IWebOperationService
    {

        private readonly IWebServiceOperationDal _webServiceOperationDal;

        public WebOperationServiceImpl(IWebServiceOperationDal webServiceOperationDal)
        {
            _webServiceOperationDal = webServiceOperationDal;
        }

        public void add(CreateWebServiceOperationRequest request)
        {
            var operation = new WebServiceOperation
            {
                Name = request.Name,
                Surname = request.Surname,
                Email = request.Email,
                PhoneNumber = request.PhoneNumber,
                Age = request.Age
            };
            _webServiceOperationDal.Add(operation);
        }

        public void delete(string name)
        {
            var operation = _webServiceOperationDal.GetAll(o => o.Name == name).FirstOrDefault();
            if (operation != null)
            {
                _webServiceOperationDal.Delete(operation);
            }
        }

        public List<WebServiceOperationResponse> getAll()
        {
            var persons = _webServiceOperationDal.GetAll();
            return persons.Select(p => new WebServiceOperationResponse
            {
                Id = p.Id,
                Name = p.Name,
                Surname = p.Surname,
                Email = p.Email,
                PhoneNumber = p.PhoneNumber,
                Age = p.Age

            }).ToList();
        }

        public List<WebServiceOperationResponse> search(string name)
        {
            var operations = _webServiceOperationDal.GetAll(o => o.Name.Contains(name));
            return operations.Select(o => new WebServiceOperationResponse
            {
                Id = o.Id,
                Name = o.Name,
                Surname = o.Surname,
                Email = o.Email,
                PhoneNumber = o.PhoneNumber,
                Age = o.Age
            }).ToList();
        }
    }
}
